src/schemas/base-helper.ts

```diff
@@ -11,10 +11,21 @@ export namespace BaseSchemaHelper {
     ) =>
         yup.object<IBaseTextBasedFieldValues>().shape({
             type: yup.string().required(),
-            label: yup.string().required(),
+            label: yup.string().required("Label is required"),
             placeholder: yup.string().optional(),
-            required: yup.boolean().required().default(false),
-            requiredErrorMsg: yup.string().optional(),
+            required: yup.boolean().required().default(true),
+            requiredErrorMsg: yup.string().test({
+                name: "requiredErrorMsg",
+                test: function (value) {
+                    if (this.parent.required && !value) {
+                        throw this.createError({
+                            message: "Error message is required",
+                            path: "requiredErrorMsg",
+                        });
+                    }
+                    return true;
+                },
+            }),
```

Comment:
actually the error message for required field is not required - there is a default error message for it already